node/core/pvf/common/src/worker.rs

```diff
@@ -93,7 +102,7 @@ pub fn worker_event_loop<F, Fut>(
 				%worker_pid,
 				"Node and worker version mismatch, node needs restarting, forcing shutdown",
 			);
-			kill_parent_node_in_emergency();
+			kill_parent_node_in_emergency(worker_pid);
```

Comment:
In a locked down worker, I don't know why we would allow this. I think this should be removed and the child at some point being locked down so that it can not kill the parent node. Just assume someone finds a way to bring down parts of the network by executing some POV.

Reply:
That's what seccomp is for! ;) This `kill_parent_node_in_emergency` is executed at the start of the child process, before any PVF threads have been spawned. 

But this made me think, kind of related to your other comment, that maybe we should do `PR_SET_PDEATHSIG` in each PVF thread as well, as right now the threads can leak. We assume the CPU timer thread kills them after a timeout, but maybe it would be good to have additional guardrails.

Reply:
Threads should be killed when the process ends. No need to do this manually.